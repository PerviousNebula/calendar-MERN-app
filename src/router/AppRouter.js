import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  BrowserRouter as Router,
  Switch,
  Redirect
} from 'react-router-dom';

import { startChecking } from '../actions/auth';

import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';
import { LoginScreen } from '../components/auth/LoginScreen';
import { CalendarScreen } from '../components/calendar/CalendarScreen';

export const AppRouter = () => {
  const { checking, uid } = useSelector(state => state.auth);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(startChecking());
  }, [dispatch]);

  if (checking) {
    return <h5>Espere...</h5>;
  }

  return (
    <Router>
      <div>
        <Switch>
          <PublicRoute
            exact
            component={ LoginScreen }
            path="/login"
            isAuthenticated={ !!uid }
          />
          <PrivateRoute
            exact
            component={ CalendarScreen }
            path="/"
            isAuthenticated={ !!uid }
          />

          <Redirect to="/" />
        </Switch>
      </div>
    </Router>
  )
}
