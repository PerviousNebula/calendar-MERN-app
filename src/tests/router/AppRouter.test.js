// React, redux imports
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

// Components
import { AppRouter } from '../../router/AppRouter';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Pruebas en <AppRouter />', () => {
  
  test('debe de mostrar el espere...', () => {
    const initState = {
      auth: {
        checking: true,
        uid: '123',
      },
    };
    let store = mockStore(initState);
    store.dispatch = jest.fn();
    
    const wrapper = mount(
      <Provider store={ store }>
        <AppRouter />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar la ruta pÃºblica', () => {
    const initState = {
      auth: {
        checking: false,
        uid: null,
      },
    };
    let store = mockStore(initState);
    store.dispatch = jest.fn();

    const wrapper = mount(
      <Provider store={ store }>
        <AppRouter />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.login-container').exists()).toBeTruthy();
  });

  test('debe de mostrar la ruta privada', () => {
    const initState = {
      auth: {
        checking: false,
        uid: '123',
        name: 'Arturo'
      },
      calendar: {
        events: [],
        activeEvent: null,
      },
      ui: {
        modalOpen: false,
      },
    };
    let store = mockStore(initState);
    store.dispatch = jest.fn();

    const wrapper = mount(
      <Provider store={ store }>
        <AppRouter />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.calendar-screen').exists()).toBeTruthy();
  });

});
