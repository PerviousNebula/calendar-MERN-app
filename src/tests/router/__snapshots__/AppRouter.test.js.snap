// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de mostrar el espere... 1`] = `
<h5>
  Espere...
</h5>
`;

exports[`Pruebas en <AppRouter /> debe de mostrar la ruta privada 1`] = `
<div>
  <div
    className="calendar-screen"
  >
    <div
      className="navbar navbar-dark bg-dark mb-4"
    >
      <span
        className="navbar-brand"
      >
        Arturo
      </span>
      <button
        className="btn btn-outline-danger"
        onClick={[Function]}
      >
        <i
          className="fas fa-sign-out-alt"
        />
        <span>
           Salir
        </span>
      </button>
    </div>
    <div
      className="rbc-calendar"
      style={
        Object {
          "height": 500,
        }
      }
    >
      <div
        className="rbc-toolbar"
      >
        <span
          className="rbc-btn-group"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            Hoy
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            &lt;
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            &gt;
          </button>
        </span>
        <span
          className="rbc-toolbar-label"
        >
          septiembre 2021
        </span>
        <span
          className="rbc-btn-group"
        >
          <button
            className="rbc-active"
            key="month"
            onClick={[Function]}
            type="button"
          >
            Mes
          </button>
          <button
            className=""
            key="week"
            onClick={[Function]}
            type="button"
          >
            Semana
          </button>
          <button
            className=""
            key="day"
            onClick={[Function]}
            type="button"
          >
            Día
          </button>
          <button
            className=""
            key="agenda"
            onClick={[Function]}
            type="button"
          >
            Agenda
          </button>
        </span>
      </div>
      <div
        aria-label="Month View"
        className="rbc-month-view"
        role="table"
      >
        <div
          className="rbc-row rbc-month-header"
          role="row"
        >
          <div
            className="rbc-header"
            key="header_0"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              lun.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_1"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              mar.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_2"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              mié.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_3"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              jue.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_4"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              vie.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_5"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              sáb.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_6"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              dom.
            </span>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  30
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  31
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  01
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  02
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  03
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  04
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  05
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  06
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  07
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  08
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  09
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  10
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  11
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  12
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  13
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  14
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  15
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  16
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  17
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  18
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  19
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg rbc-today"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell rbc-now rbc-current"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  20
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  21
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  22
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  23
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  24
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  25
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  26
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  27
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  28
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  29
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  30
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  01
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  02
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  03
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="btn btn-primary fab"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fas fa-plus"
      />
    </button>
    <Portal
      containerInfo={
        <div
          class="ReactModalPortal"
        />
      }
    />
  </div>
</div>
`;

exports[`Pruebas en <AppRouter /> debe de mostrar la ruta pública 1`] = `
<div>
  <div
    className="container login-container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-6 login-form-1"
      >
        <h3>
          Ingreso
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="lEmail"
              onChange={[Function]}
              placeholder="Correo"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="lPassword"
              onChange={[Function]}
              placeholder="Contraseña"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="btnSubmit"
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
      <div
        className="col-md-6 login-form-2"
      >
        <h3>
          Registro
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="rName"
              onChange={[Function]}
              placeholder="Nombre"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="rEmail"
              onChange={[Function]}
              placeholder="Correo"
              type="email"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="rPassword1"
              onChange={[Function]}
              placeholder="Contraseña"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="rPassword2"
              onChange={[Function]}
              placeholder="Repita la contraseña"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="btnSubmit"
              type="submit"
              value="Crear cuenta"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
